### PR TITLE
matplotlib-cpp/cci.20210111

### PR DESCRIPTION
**Pull request by blackliner**
Specify library name and version:  **matplotlib-cpp/cci.20210111**

Looking for feedback regarding the `python 3` requirement, do we have a system package?

Requesting issue: https://github.com/conan-io/conan-center-index/issues/3033

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/matplotlib-cpp/all/conandata.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+sources:
+  cci.20200422:
+    - url: https://raw.githubusercontent.com/lava/matplotlib-cpp/d74105036a82e6ccede3fa1a7d872ac1677b00ab/matplotlibcpp.h
+      sha256: 913638b1d03ccfe28f58f327a8c1247aec943f88f19c37648fd01e239420fb90
+    - url: https://raw.githubusercontent.com/lava/matplotlib-cpp/d74105036a82e6ccede3fa1a7d872ac1677b00ab/LICENSE
+      sha256: d0dc4eae3be039d576e127912d16f78a7ba4e0e6205cd75103efd1a45b89a4ba
```
Comment (SpaceIm) [inline recipes/matplotlib-cpp/all/conandata.yml:6]: Is it not less maintenance effort with https://github.com/lava/matplotlib-cpp/archive/d74105036a82e6ccede3fa1a7d872ac1677b00ab.zip?

**Comment by SpaceIm** [comment]
Can't be added right now, it requires python recipe.

File: recipes/matplotlib-cpp/all/conandata.yml (hunk 1 above)
Comment (SpaceIm) [reply recipes/matplotlib-cpp/all/conandata.yml:6]: Why this "old" commit by the way?

Comment (blackliner) [reply recipes/matplotlib-cpp/all/conandata.yml:6]: Uh, how did you get that, did not see a release :-/

Comment (SpaceIm) [reply recipes/matplotlib-cpp/all/conandata.yml:6]: github provides a way to download a zip of any commit.

Comment (blackliner) [reply recipes/matplotlib-cpp/all/conandata.yml:6]: > Why this "old" commit by the way?

I `git bisected` this version to be the last one that supports C++11, followups need C++17

**Comment by blackliner** [comment]
> Can't be added right now, it requires cpython recipe: #1510

Don't want to start a big discussion, but can't we assume with certainty that python is available? I mean, no python, no conan?

**Comment by SpaceIm** [comment]
> 
> 
> > Can't be added right now, it requires cpython recipe: #1510
> 
> Don't want to start a big discussion, but can't we assume with certainty that python is available? I mean, no python, no conan?

It depends how it was installed. python might not be available (conan windows installer for example).
Moreover:
- consumers might be surprised to be "polluted" with numpy in their system installed python.
- no control on python version

File: recipes/matplotlib-cpp/all/conandata.yml (hunk 1 above)
Comment (blackliner) [reply recipes/matplotlib-cpp/all/conandata.yml]: done (the download part)

**Comment by blackliner** [comment]
> It depends how it was installed. python might not be available (conan windows installer for example).
> Moreover:
> 
> * consumers might be surprised to be "polluted" with numpy in their system installed python.

Fully agree, just curious to see if that actually works on CCI-CI 🙈 

> * no control on python version

For this package, I do not really care, but I get the point, for example if you need 3.x

The PR you mentioned tries to make python available from scratch, why not start wiht a `python/system@` packages first?

**Comment by uilianries** [comment]
It requires [numpy](https://numpy.org/doc/stable/reference/c-api/coremath.html) first.

**Comment by SpaceIm** [comment]
And cpython recipe.

**Comment by blackliner** [comment]
So, long story short, its currently impossible to build packages that depend on python?

Would depending on `python/system` be an alternative?

**Comment by blackliner** [comment]
@uilianries @SpaceIm : Is it possible at all to depend on python and/or numpy in the current state of the CCI? Are there any other packages that do something similar? I don't know if it makes sense to grab another copy of python and numpy for this package, I think its safe to just check if those packages exist on the users system, and error out if that is not the case.

But how should one tell this to the CCI-CI ?

**Comment by SpaceIm** [comment]
Currently, several recipes implicitly rely on python as a build requirement. But for matplotlib-cpp, it's worse, it's a requirement, it links to python library.

**Comment by blackliner** [comment]
> Currently, several recipes implicitly rely on python as a build requirement. But for matplotlib-cpp, it's worse, it's a requirement, it links to python library.

So, no luck for now?

**Comment by SpaceIm** [comment]
Yes, "Welcome to matplotlib-cpp, **possibly the simplest C++ plotting library**" is not true for conan 😞

**Pull request closed without merging**